controller/workitem.go

```diff
@@ -719,6 +741,19 @@ func ConvertWorkItem(request *http.Request, wit workitem.WorkItemType, wi workit
 	return op, nil
 }
 
+// addNewFieldNames adds new field names (system_) to the payload. The payload
+// contains system.* and system_* field names with the same value
+// TODO (ibrahim) - remove this once field name migration is completed
+func addNewFieldNames(wi *app.WorkItem) {
+	attributeList := wi.Attributes
+	for newAttributeName := range attributeList {
+		oldAttributeName, ok := workitem.NewToOldFieldNameMap[newAttributeName]
+		if ok {
```

Comment:
This essentially means, only overwrite the value given under the old name, if a new value under the new name is given, right?

Reply:
No no. It means that if we have an `old name` for the `new attribute` name we add it to the `Attributes` map. We don't overwrite anything.

Reply:
Well, you are setting `wi.Attributes[oldAttributeName]`. If that did exist before, you're overwriting it with the value as it is stored under the new name, no?

Effectively:

```go
oldAttributeName, ok := workitem.NewToOldFieldNameMap["system_title"]

// now oldAttributeName will be "system.title"

if ok {
    wi.Attributes["system.title"] = wi.Attributes["system_title"]
}
```

Reply:
Yes. That's correct. But the other thing to note is `wi.Attributes["system.title"]` will always be empty.
We use `replaceFieldNames` method in `convertJSONtoWorkitem` method, which replaces all the `system.` fields
we know about and `addFieldNames` method in `ConvertWorkitemToJSON` that adds all the `system.` fields we know about.